lib/AST/ProtocolConformance.cpp

```diff
@@ -241,7 +241,7 @@ void NormalProtocolConformance::setTypeWitness(
   assert(getProtocol() == cast<ProtocolDecl>(assocType->getDeclContext()) &&
          "associated type in wrong protocol");
   assert(TypeWitnesses.count(assocType) == 0 && "Type witness already known");
-  assert(!isComplete() && "Conformance already complete?");
+  assert((!isComplete() || isInvalid()) && "Conformance already complete?");
```

Comment:
"Well actually" (sorry 😬), it would have to be "Conformance already complete **and not** invalid". Which to me seems implied by the fact that the assertion is being hit. If it's invalid we don't care whether it's complete.